CgfConverter/CryEngineCore/Chunks/ChunkController_905.cs

```diff
@@ -0,0 +1,415 @@
+
+    public override void Read(BinaryReader b)
+    {
+        base.Read(b);
+
+        // TODO: detect shit
+        ((EndiannessChangeableBinaryReader) b).IsBigEndian = true;
+
+        NumKeyPos = b.ReadUInt32();
+        NumKeyRot = b.ReadUInt32();
+        NumKeyTime = b.ReadUInt32();
+        NumAnims = b.ReadUInt32();
+
+        var keyTimeLengths = Enumerable.Range(0, (int) NumKeyTime).Select(_ => b.ReadUInt16()).ToList();
+        var keyTimeFormats = Enumerable.Range(0, (int) EKeyTimesFormat.eBitset + 1).Select(_ => b.ReadUInt32()).ToList();
+        
+        var keyPosLengths = Enumerable.Range(0, (int) NumKeyPos).Select(_ => b.ReadUInt16()).ToList();
+        var keyPosFormats = Enumerable.Range(0, (int) ECompressionFormat.eAutomaticQuat).Select(_ => b.ReadUInt32()).ToList();
+
+        var keyRotLengths = Enumerable.Range(0, (int) NumKeyRot).Select(_ => b.ReadUInt16()).ToList();
+        var keyRotFormats = Enumerable.Range(0, (int) ECompressionFormat.eAutomaticQuat).Select(_ => b.ReadUInt32()).ToList();
+
+        var keyTimeOffsets = Enumerable.Range(0, (int) NumKeyTime).Select(_ => b.ReadUInt32()).ToList();
+        var keyPosOffsets = Enumerable.Range(0, (int) NumKeyPos).Select(_ => b.ReadUInt32()).ToList();
+        var keyRotOffsets = Enumerable.Range(0, (int) NumKeyRot).Select(_ => b.ReadUInt32()).ToList();
+        var trackLength = b.ReadUInt32();
+        
+        Debug.Assert(keyTimeOffsets.All(x => (x & 3) == 0));
+        Debug.Assert(keyPosOffsets.All(x => (x & 3) == 0));
+        Debug.Assert(keyRotOffsets.All(x => (x & 3) == 0));
+        Debug.Assert((trackLength & 3) == 0);
+        
+        keyRotOffsets.Add(trackLength);
+        keyPosOffsets.Add(keyRotOffsets.First());
+        keyTimeOffsets.Add(keyPosOffsets.First());
+
+        var trackOffset = b.BaseStream.Position;
+        if ((trackOffset & 3) != 0)
+            trackOffset = (trackOffset & ~3) + 4; 
+
+        KeyTimes = new List<List<float>>();
+        foreach (var (length, from) in keyTimeLengths.Zip(keyTimeOffsets))
+        {
+            b.BaseStream.Position = trackOffset + from;
+
+            List<float> data;
+            if (keyTimeFormats[(int)EKeyTimesFormat.eF32] > 0)
+            {
+                --keyTimeFormats[(int)EKeyTimesFormat.eF32];
+                data = Enumerable.Range(0, length).Select(_ => b.ReadSingle()).ToList();
+            }
+            else if (keyTimeFormats[(int)EKeyTimesFormat.eUINT16] > 0)
+            {
+                --keyTimeFormats[(int)EKeyTimesFormat.eUINT16];
+                data = Enumerable.Range(0, length).Select(_ => (float)b.ReadUInt16()).ToList();
+            }
+            else if (keyTimeFormats[(int)EKeyTimesFormat.eByte] > 0)
+            {
+                --keyTimeFormats[(int)EKeyTimesFormat.eByte];
+                data = b.ReadBytes(length).Select(x => (float) x).ToList();
+            }
+            else if (keyTimeFormats[(int)EKeyTimesFormat.eF32StartStop] > 0)
+            {
```

Comment:
For the Not Implemented if statements, you can get rid of the squigglies (`{` and `}`) to save some vertical space.